dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckSecurityOwnershipValidationHook.java

```diff
@@ -206,10 +201,42 @@ public void validateEnrollment( ValidationErrorReporter reporter, Enrollment enr
             }
         }
 
-        checkWriteEnrollmentAccess( reporter, enrollment, program, enrollment.getTrackedEntity(),
+        checkWriteEnrollmentAccess( reporter, enrollment, program, context,
             ownerOrgUnit );
     }
 
+    private void checkEnrollmentOrgUnit( Enrollment enrollment, TrackerImportValidationContext context,
```

Comment:
reporter usually goes first in these checks, I would stick to that.

I would go for this order `reporter, context, strategy, enrollment, program)` (grouping the "entities" like enrollment, program and the other types)